File: lms-video-streaming-fe/src/services/profile.service.ts
import axiosClient from "../config/axiosClient.config";
import type { ResponseData } from "../types/common.types";
import type {
  UserProfileResponse,
  ProfileUpdatingRequest,
  UserCourseResponse,
} from "../types/user.types";

export const profileService = {
  getMe: async (): Promise<ResponseData<UserProfileResponse>> => {
    return axiosClient.get("/profile/me");
  },

  updateProfile: async (
    data: ProfileUpdatingRequest,
  ): Promise<ResponseData> => {
    return axiosClient.post("/profile/me", data);
  },

  uploadAvatar: async (file: File): Promise<ResponseData<string>> => {
    const formData = new FormData();
    formData.append("file", file);

    return axiosClient.post("/profile/upload", formData, {
      headers: {
        "Content-Type": "multipart/form-data",
      },
    });
  },

  getCourseMe: async (): Promise<ResponseData<UserCourseResponse[]>> => {
    return axiosClient.get("/profile/course-me");
  },
};

File: server/lms-server-web/src/main/java/com/hust/lms/streaming/service/ProfileService.java
package com.hust.lms.streaming.service;

import com.hust.lms.streaming.dto.request.user.ProfileUpdatingRequest;
import com.hust.lms.streaming.dto.response.user.UserCourseResponse;
import com.hust.lms.streaming.dto.response.user.UserProfileResponse;
import java.util.List;
import org.springframework.web.multipart.MultipartFile;

public interface ProfileService {
  String upload(MultipartFile file);
  void profile(ProfileUpdatingRequest request);
  UserProfileResponse me();
  List<UserCourseResponse> getCourseMe();
}
